### PR TITLE
manifest: Update to TF-M v1.8

### PR DESCRIPTION
**Pull request by mswarowsky**
Upmerge to update the TF-M version to 1.8

Ref: NCSDK-22275

test-sdk-nrf: sdk-nrf-pr-12139

### CODE REVIEW

**Comment by doublemis1** [comment]
Matter Wifi sample failed on connecting to WiFi

```
I: 18116 [DL]Wi-Fi scan done (0)
E: ***** BUS FAULT *****
E:   Precise data bus error
E:   BFAR Address: 0x26902bb
E: r0/a1:  0x026902bb  r1/a2:  0x000e47b2  r2/a3:  0x000000a0
E: r3/a4:  0x00000077 r12/ip:  0x00000000 r14/lr:  0x00099f4b
E:  xpsr:  0x21100000
E: s[ 0]:  0x00000000  s[ 1]:  0x00000000  s[ 2]:  0x00000000  s[ 3]:  0x00000000
E: s[ 4]:  0x00000000  s[ 5]:  0x00000000  s[ 6]:  0x00000000  s[ 7]:  0x00000000
E: s[ 8]:  0x00000000  s[ 9]:  0x00000000  s[10]:  0x00000000  s[11]:  0x00000000
E: s[12]:  0x00000000  s[13]:  0x00000000  s[14]:  0x00000000  s[15]:  0x00000000
E: fpscr:  0x20018b24
E: Faulting instruction address (r15/pc): 0x0000f648
E: >>> ZEPHYR FATAL ERROR 25: Unknown error on CPU 0
E: Current thread: 0x2000e160 (CHIP)
E: Halting system
```

File: modules/tfm/tfm/boards/common/crypto_keys.c [hunk 1]
```diff
@@ -17,11 +16,8 @@
 #include <hw_unique_key.h>
 #include <identity_key.h>
 
-#define TFM_NS_PARTITION_ID (-1)
-
-#ifndef MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER
-#error "MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER must be selected in Mbed TLS config file"
-#endif
+#define NUMBER_OF_ELEMENTS_OF(x) sizeof(x)/sizeof(*x)
```
Comment (vili-nordic) [inline modules/tfm/tfm/boards/common/crypto_keys.c:19]: Is the ARRAY_SIZE macro usable here instead?

**Review by vili-nordic — APPROVED** [review]
looks fine to me

**Comment by doublemis1** [comment]
> Matter Wifi sample failed on connecting to WiFi
> 
> ```
> I: 18116 [DL]Wi-Fi scan done (0)
> E: ***** BUS FAULT *****
> E:   Precise data bus error
> E:   BFAR Address: 0x26902bb
> E: r0/a1:  0x026902bb  r1/a2:  0x000e47b2  r2/a3:  0x000000a0
> E: r3/a4:  0x00000077 r12/ip:  0x00000000 r14/lr:  0x00099f4b
> E:  xpsr:  0x21100000
> E: s[ 0]:  0x00000000  s[ 1]:  0x00000000  s[ 2]:  0x00000000  s[ 3]:  0x00000000
> E: s[ 4]:  0x00000000  s[ 5]:  0x00000000  s[ 6]:  0x00000000  s[ 7]:  0x00000000
> E: s[ 8]:  0x00000000  s[ 9]:  0x00000000  s[10]:  0x00000000  s[11]:  0x00000000
> E: s[12]:  0x00000000  s[13]:  0x00000000  s[14]:  0x00000000  s[15]:  0x00000000
> E: fpscr:  0x20018b24
> E: Faulting instruction address (r15/pc): 0x0000f648
> E: >>> ZEPHYR FATAL ERROR 25: Unknown error on CPU 0
> E: Current thread: 0x2000e160 (CHIP)
> E: Halting system
> ```

Issue connected with regression on main.
Should be already fixed

File: modules/tfm/tfm/boards/common/crypto_keys.c (hunk 1 above)
Comment (mswarowsky) [reply modules/tfm/tfm/boards/common/crypto_keys.c:19]: Oh wasn't aware of that macro and just used what was used in other TF-M platforms. changed it to `ARRAY_SIZE`

File: modules/tfm/tfm/boards/common/crypto_keys.c [hunk 2]
```diff
@@ -81,144 +76,71 @@ static enum tfm_plat_err_t tfm_plat_get_iak(uint8_t *buf, size_t buf_len,
...
-			TFM_BUILTIN_KEY_LOADER_KEY_LOCATION);
-		break;
-#endif /* CONFIG_HW_UNQUE_KEY */
 
+/**
+ * @brief Table describing per-key user policies
+ *
+ */
+static const tfm_plat_builtin_key_policy_t g_builtin_keys_policy[] = {
+    {.key_id = TFM_BUILTIN_KEY_ID_HUK, .per_user_policy = 0, .usage = PSA_KEY_USAGE_DERIVE},
```
Comment (joerchan) [inline modules/tfm/tfm/boards/common/crypto_keys.c:107]: This should be between #ifdef CONFIG_HW_UNIQUE_KEY, to be consistent with the rest.

File: include/tfm/platform_builtin_key_loader_ids.h [hunk 3]
```diff
@@ -11,7 +11,7 @@
 extern "C" {
 #endif
 
-#define TFM_BUILTIN_MAX_KEY_LEN 96
+#define TFM_BUILTIN_MAX_KEY_LEN 32
```
Comment (joerchan) [inline include/tfm/platform_builtin_key_loader_ids.h:14]: Appears we may have been wasting bytes before.

File: subsys/nrf_security/configs/psa_crypto_config.h.template [hunk 4]
```diff
@@ -221,7 +221,9 @@
 
 /* TF-M */
 #cmakedefine MBEDTLS_PSA_CRYPTO_SPM
+#ifndef MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER
```
Comment (joerchan) [inline subsys/nrf_security/configs/psa_crypto_config.h.template:224]: I'm somewhat supsicious about this. Is this defined to something elsewhere?

**Review by joerchan — CHANGES_REQUESTED** [review]

File: include/tfm/platform_builtin_key_loader_ids.h (hunk 3 above)
Comment (mswarowsky) [reply include/tfm/platform_builtin_key_loader_ids.h:14]: discussed offline

File: modules/tfm/tfm/boards/common/crypto_keys.c (hunk 2 above)
Comment (mswarowsky) [reply modules/tfm/tfm/boards/common/crypto_keys.c:107]: Agree

File: modules/tfm/tfm/boards/common/crypto_keys.c [hunk 5]
```diff
@@ -81,144 +76,71 @@ static enum tfm_plat_err_t tfm_plat_get_iak(uint8_t *buf, size_t buf_len,
...
-	if (err != PSA_SUCCESS) {
-		return TFM_PLAT_ERR_SYSTEM_ERR;
-	}
+    {.key_id = TFM_BUILTIN_KEY_ID_IAK,
+     .slot_number = TFM_BUILTIN_KEY_SLOT_IAK,
+     .lifetime = TFM_BUILTIN_KEY_LOADER_LIFETIME,
+     .loader_key_func = tfm_plat_get_iak},
 #endif /* TFM_PARTITION_INITIAL_ATTESTATION */
-#endif /* CONFIG_HW_UNIQUE_KEY || TFM_PARTITION_INITIAL_ATTESTATION */
+    {},
```
[the remark this answers is not in the document]
Comment (mswarowsky) [reply modules/tfm/tfm/boards/common/crypto_keys.c]: discussed offline

File: subsys/nrf_security/configs/psa_crypto_config.h.template (hunk 4 above)
Comment (mswarowsky) [reply subsys/nrf_security/configs/psa_crypto_config.h.template:225]: discussed offline

**Review by joerchan — APPROVED** [review]

**Pull request merged**